src/common/transactions/transaction-utils.ts

```diff
@@ -119,3 +138,107 @@ export const generateSTXTransferTx = ({
 
 export const stacksTransactionToHex = (transaction: StacksTransaction) =>
   `0x${transaction.serialize().toString('hex')}`;
+
+export function txHasTime(tx: Tx) {
+  return !!(
+    ('burn_block_time_iso' in tx && tx.burn_block_time_iso) ||
+    ('parent_burn_block_time_iso' in tx && tx.parent_burn_block_time_iso)
+  );
+}
+
+export const isAddressTransactionWithTransfer = (
+  transaction: AddressTransactionWithTransfers | Tx
+) => 'tx' in transaction;
```

Comment:
```suggestion
export function isAddressTransactionWithTransfer (
  transaction: AddressTransactionWithTransfers | Tx
): transaction is AddressTransactionWithTransfers {
  return 'tx' in transaction;
}
```

We can give the type system more information here, which I think means we can avoid the type casting in the component. Also formatted as function declaration in line with rest of file.